### PR TITLE
FirstCommit

### PR DESCRIPTION
**Pull request by AlinBivolan**

### CODE REVIEW

File: src/Display/Display.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import classes from "./Display.module.css";
+
+function Display() {
+  return (
+    <div>
+        <div className={classes.container}>
+        </div>
```
Comment (andreipop02) [inline src/Display/Display.js]: make this self-close tag, as you dont wrap anything

File: src/TextInput/EmojiModal.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import { useRef } from "react";
+import React from "react";
```
Comment (andreipop02) [inline src/TextInput/EmojiModal.js]: make a single import for those 2, they are both from "react"

**Review by andreipop02 — CHANGES_REQUESTED** [review]

File: src/TextInput/Input.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  };
+  const addEmojiFromRecent = (emoji) => {
+    setTextValue(textValue + emoji);
+  };
+
+  const emojiHandler = () => {
+    setEmoji(true);
+  };
+
+  const modalHandler = () => {
```
Comment (andreipop02) [inline src/TextInput/Input.js]: name it more specifically ( something like: closeEmojiModal )

File: src/TextInput/Input.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    if (!recentEmoji.includes(emoji)) {
+      setRecentEmoji([...recentEmoji.splice(1, 17), emoji]);
+    }
+    setTextValue(textValue + emoji);
+  };
+  const addEmojiFromRecent = (emoji) => {
+    setTextValue(textValue + emoji);
+  };
+
+  const emojiHandler = () => {
```
Comment (andreipop02) [inline src/TextInput/Input.js]: name it more specifically ( something like: openEmojiModal )

File: src/constants/arrays.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+  "🐹",
+  "🍪",
+  "💀",
+  "📞",
+  "☠️",
+  "🎤",
+  "🔨",
+];
+
+// export const Category = [
```
Comment (catalinracz) [inline src/constants/arrays.js]: remove if not used anymore

File: src/TextInput/ModalContent/Categories.js [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+        onChange={(e) => handlerFilter(e.target.value)}
+        type="type"
+        placeholder="Search"
+        className={classes.search}
+        value={value}
+      />
+      <span ref={emojis} className={classes.emoji_container}>
+        {value === "" ? (
+          <>
+            <h1 className={classes.category_name}>Recent Used</h1>
```
Comment (andreipop02) [inline src/TextInput/ModalContent/Categories.js]: Add this to a strings file and import from there (ex: strings.recentUsed)

File: src/TextInput/ModalContent/Emoji.js [hunk 7]
```diff
@@ -0,0 +1,30 @@
+import React from "react";
+
+import classes from "./Emoji.module.css";
+import emoji from "../../constants/emoji.json";
+
+const Emoji = ({ category, changeValue, i, categorySection }) => {
+  return (
+    <>
+      <h1 className={classes.category_name} ref={categorySection.current[i]}>
+        {category}{" "}
```
Comment (andreipop02) [inline src/TextInput/ModalContent/Emoji.js]: What do we need "{" "}" for?

File: src/TextInput/Input.js [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+          onConfirm={modalHandler}
+          changeValue={addEmoji}
+          changeValueRecent={addEmojiFromRecent}
+          recent={recentEmoji}
+        />
+      )}
+      <textarea
+        onChange={(e) => setTextValue(e.target.value)}
+        className={classes.display}
+        placeholder="Your Messages..."
```
Comment (andreipop02) [inline src/TextInput/Input.js]: add to strings file

**Review by catalinracz — CHANGES_REQUESTED** [review]

**Review by andreipop02 — APPROVED** [review]

**Comment by catalinracz** [comment]
Screenshot 2022-06-16 at 4 10 10 pm
The picker comes over the input on smaller screens.

**Pull request merged**